scripts/increase_version.py

```diff
@@ -0,0 +1,593 @@
+                              "such as alembic.")
+
+    rdb_ver = rdb_ver.add_mutually_exclusive_group(required=False)
+
+    rdb_ver.add_argument('--rdb-major',
+                         dest='rdb_ver',
+                         action='store_const',
+                         const='major',
+                         default=argparse.SUPPRESS,
+                         help="Increase the major version. This mode will "
+                              "set 'minor' to 0.")
+
+    rdb_ver.add_argument('--rdb-minor',
+                         dest='rdb_ver',
+                         action='store_const',
+                         const='minor',
+                         default=argparse.SUPPRESS,
+                         help="Increase the minor version.")
+
+    def __handle(args):
+        """
+        Custom argument handler for this script to give invocation-related
+        error messages and apply extra rules that can not be expressed
+        with argparse.
+        """
+
+        def arg_match(options):
+            """
+            Checks and selects the option strings specified in 'options'
+            that are present in the invocation.
+            """
+
+            matched_args = []
+            for option in options:
+                if any([arg if option.startswith(arg) else None
+                        for arg in sys.argv[1:]]):
+                    matched_args.append(option)
+                    continue
+
+            return matched_args
+
+        if args.cc_ver == 'major':
+            # CodeChecker major version increase automatically implies a
+            # major version increase on everything else.
+
+            minor_opts = ['--api-minor', '--pdb-minor', '--rdb-minor']
+            matching = arg_match(minor_opts)
+            if any(matching):
+                first_matching = next(iter([match for match in matching]))
+                parser.error("argument {0}: not allowed with argument "
+                             "--major".format(first_matching))
+                # parser.error() terminates with return code 2.
+
+            setattr(args, 'api_ver', 'major')
+            setattr(args, 'pdb_ver', 'major')
+            setattr(args, 'rdb_ver', 'major')
+
+        __main__(args)
+
+    parser.set_defaults(func=__handle)
+    return parser
+
+
+def increase_version(version_dict, attribute):
+    new_value = str(int(version_dict[attribute]) + 1)
+    version_dict[attribute] = new_value
+
+
+def format_version2(version):
+    return "{0}.{1}".format(version['major'],
+                            version['minor'])
+
+
+def split_version2(version):
+    parts = version.split('.')
+    return {'major': parts[0], 'minor': parts[1]}
+
+
+def format_version3(version):
+    return "{0}.{1}.{2}".format(version['major'],
+                                version['minor'],
+                                version['revision'])
+
+
+def __make_alembic_revision(config, message):
```

Comment:
I'm not sure we should add the alembic call here to generate the migration file.
A warning should be added to check if there is a migration script.
In many cases it will be auto generated manually against an existing database.